### DISCUSSION TITLE
Storybook composition SSR

### DISCUSSION THREAD

**Discussion by daraccrafter** [storybookjs/storybook#22649] [Help]
I am wondering if there is a possibility to have the refs urls be server side rendered. My plan is to run a public composed storybook on a kubernetes cluster. The storybook would be composed out of different projects storybooks which are all running on the cluster and are exposed as services. But the service urls cannot be found since they are only accessible from within the cluster and the composed storybook renders the refs on the client. Im wondering if there is a way to solve this but still keep the non composed storybooks unaccessible from the outside.

**Comment by shilman** [comment]
I'm not sure I understand the issue. What happens when you reference these internal URLs on the client side? I'd expect that the composed storybook would work for anybody whose browsers could resolve those URLs and fail for anybody else, which seems like what you're after?

**Reply by daraccrafter** [reply]
The thing is the service urls are not resolvable on the client. These service urls are only resolvable from the cluster. The whole composition originates from the browser.
Example:
If i try to load my public storybook **storybook.example.com** it tries to fetch **http://myservice:3000** storybook but it cant because that url is only resolvable internally inside the cluster.
